src/js/api/api.js

```diff
@@ -388,8 +302,17 @@ define([
             plugins.registerPlugin(id, target, arg1, arg2);
         };
 
+        function _forwardStateEvent(evt) {
+            _this.trigger(evt.newstate, evt);
+        }
         _this.setController = function (player) {
+            if (_controller) {
+                _controller.off('all', _this.trigger);
+                _controller.off(events.JWPLAYER_PLAYER_STATE, _forwardStateEvent);
```

Comment:
We discussed this in person. 

The question is if we can safely assume that all references to the previous controller are dropped, and if so it would be safe to use .off().

However if the controller may be used again (perhaps it is swapped out due to a plugin like instream) then this would break other functionality.

For now, we are going to leave the conservative decision.